packages/host/app/services/matrix-sdk-loader.ts

```diff
@@ -133,6 +135,70 @@ async function createRealmSession(
 
   return await realmAuthClient.getJWT();
 }
+async function setAccountData<T>(
+  this: ExtendedClient,
+  fetch: typeof globalThis.fetch,
+  type: string,
+  data: T,
+) {
+  let userId = this.getUserId();
+  if (!userId) {
+    throw new Error(`Cannot set account data without matrix session`);
+  }
+  let response = await fetch(
+    `${this.baseUrl}/_matrix/client/v3/user/${encodeURIComponent(
+      userId,
+    )}/account_data/${type}`,
+    {
+      method: 'PUT',
+      headers: {
+        Authorization: `Bearer ${this.getAccessToken()}`,
+      },
+      body: JSON.stringify(data),
+    },
+  );
+  if (!response.ok) {
+    let json = await response.json();
+    throw new Error(
+      `Unable to set account data '${type}' for ${this.getUserId()}: status ${
+        response.status
+      } - ${JSON.stringify(json)}`,
+    );
+  }
+}
+
+async function getAccountData<T>(
+  this: ExtendedClient,
+  fetch: typeof globalThis.fetch,
+  type: string,
+) {
+  let userId = this.getUserId();
+  if (!userId) {
+    throw new Error(`Cannot get account data without matrix session`);
+  }
+  let response = await fetch(
+    `${this.baseUrl}/_matrix/client/v3/user/${encodeURIComponent(
+      userId,
+    )}/account_data/${type}`,
+    {
+      headers: {
+        Authorization: `Bearer ${this.getAccessToken()}`,
+      },
+    },
+  );
+  if (response.status === 404) {
+    return;
+  }
+  let json = await response.json();
+  if (!response.ok) {
+    throw new Error(
+      `Unable to get account data '${type}' for ${this.getUserId()}: status ${
+        response.status
+      } - ${JSON.stringify(json)}`,
+    );
+  }
+  return json as T;
+}
```

Comment:
What context are these used in? I’m sure I’m missing a part in my mental model because the Matrix client has this built in, right?

Reply:
This is in the host app, the matrix js SDK is missing this

Reply:
The matrix client has `getAccountDataFromServer`, which I belive does this.